src/main/java/forum/dao/ThreadDAOImpl.java

```diff
@@ -0,0 +1,40 @@
+package forum.dao;
+
+import java.util.List;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import forum.entity.Thread;
+
+@Repository
+public class ThreadDAOImpl implements ThreadDAO {
+	
+	@Autowired
+	private SessionFactory sessionFactory;
+
+	@Override
+	public void saveThread(Thread thread) {
+		
+		Session currentSession = sessionFactory.getCurrentSession();
+		
+		currentSession.saveOrUpdate(thread);
```

Comment:
Could we use only save here? Imo it is confusing use `saveOrUpdate` in this face :D

Reply:
In the future if we need to be able to update a thread we could use this existing method for it, right? I can make it use only save for now and change it in the future or just make a different method for it if you think it's better that way.